### PR TITLE
Disallow duplicate phone and email to exist

### PR DESCRIPTION
**Pull request by Ma-Xizi**

### CODE REVIEW

File: src/test/java/tutorease/address/logic/commands/AddContactCommandTest.java [hunk 1]
```diff
@@ -142,6 +142,16 @@ public boolean hasPerson(Person person) {
...
+        @Override
+        public boolean hasSamePhone(Person person) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public boolean hasSameEmail(Person person) {
+            throw new AssertionError("This method should not be called.");
+        }
+
```
Comment (Chronoxy) [inline src/test/java/tutorease/address/logic/commands/AddContactCommandTest.java:154]: I like the catch you did on how these methods should not be called within the other commands

**Review by Chronoxy — APPROVED** [review]
LGTM 👍

**Review by NorbertLoh — COMMENTED** [review]
Nits to fix. Otherwise LGTM!

File: src/main/java/tutorease/address/logic/commands/AddContactCommand.java [hunk 2]
```diff
@@ -61,6 +65,14 @@ public CommandResult execute(Model model) throws CommandException {
...
         }
 
+        if (model.hasSamePhone(toAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PHONE);
+        }
+
+        if (model.hasSameEmail(toAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_EMAIL);
+        }
+
```
Comment (NorbertLoh) [inline src/main/java/tutorease/address/logic/commands/AddContactCommand.java:75]: Perhaps can separate these to adhere to SLAP.

**Review by NorbertLoh — COMMENTED** [review]
Nits to fix. Otherwise LGTM!

File: src/main/java/tutorease/address/logic/commands/AddContactCommand.java (hunk 2 above)
Comment (Ma-Xizi) [reply src/main/java/tutorease/address/logic/commands/AddContactCommand.java]: Fixed!

**Review by NorbertLoh — APPROVED** [review]
LGTM!

**Pull request merged**